resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceRequestHelperSuite.scala

```diff
@@ -0,0 +1,207 @@
+    sparkConf.set(EXECUTOR_CORES.key, "2")
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("memory defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_MEMORY, "30G")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_EXECUTOR_MEMORY)
+  }
+
+  test("cores defined with new config for executor") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_EXECUTOR_CORES, "5")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_EXECUTOR_CORES)
+  }
+
+  test("memory defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_AM_MEMORY, "1G")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_AM_MEMORY)
+  }
+
+  test("memory defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_MEMORY, "1G")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_DRIVER_MEMORY)
+  }
+
+  test("cores defined with new config, client mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_AM_CORES, "3")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_AM_CORES)
+  }
+
+  test("cores defined with new config for driver, cluster mode") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(NEW_CONFIG_DRIVER_CORES, "1G")
+    verifyValidateResourcesException(sparkConf, NEW_CONFIG_DRIVER_CORES)
+  }
+
+  test("various duplicated definitions") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "2G")
+    sparkConf.set(DRIVER_CORES.key, "2")
+    sparkConf.set(EXECUTOR_MEMORY.key, "2G")
+    sparkConf.set(EXECUTOR_CORES.key, "4")
+    sparkConf.set(AM_MEMORY.key, "3G")
+    sparkConf.set(NEW_CONFIG_EXECUTOR_MEMORY, "3G")
+    sparkConf.set(NEW_CONFIG_AM_MEMORY, "2G")
+    sparkConf.set(NEW_CONFIG_DRIVER_MEMORY, "2G")
+
+    val thrown = intercept[SparkException] {
+      ResourceRequestHelper.validateResources(sparkConf)
+    }
+    thrown.getMessage should (
+      include(NEW_CONFIG_EXECUTOR_MEMORY) and
+      include(NEW_CONFIG_AM_MEMORY) and
+      include(NEW_CONFIG_DRIVER_MEMORY))
+  }
+
+  private def verifySetResourceRequestsSuccessful(definedResourceTypes: List[String],
+      resourceRequests: Map[String, String],
+      expectedResources: Map[String, ResourceInformation]) = {
+    assume(ResourceRequestHelper.isYarnResourceTypesAvailable())
+    ResourceRequestTestHelper.initializeResourceTypes(definedResourceTypes)
+
+    val resource = createResource
+    ResourceRequestHelper.setResourceRequests(resourceRequests, resource)
+
+    expectedResources.foreach { case (name, ri) =>
+      val resourceInfo: ResourceInformation = ResourceRequestTestHelper
+          .getResourceInformationByName(resource, name)
+      assert(resourceInfo === ri)
+    }
+  }
+
+  private def verifySetResourceRequestsExceptionWithNullRequest(definedResourceTypes: List[String],
```

Comment:
arg in next line. also in a lot of methods you added below.

This particular method is also called from a single place, so it doesn't need to exist. Please check that's not the case with others.

Reply:
Fixed, actually removed two methods that were used just from one place.